### ISSUE TITLE
The variable `GOMAXPROCS` does not work properly in Container environment

### ISSUE THREAD

**Issue by DavadDi** [prometheus/prometheus#6413]
In Kubernetes env, the containers spec as follows: 
```
  containers:
  - args:
    image: quay.io/prometheus/prometheus:v2.11.0
    imagePullPolicy: IfNotPresent
    resources:
      limits:
        cpu: "4" 
        memory: 128Gi
      requests:
        cpu: "4"
        memory: 128Gi
```

In Runtime Information of Prometheus:

```
GOMAXPROCS:  96   // The host CPUs number, I'd like to see the values of `containers.requests` 4
```

In this case, the Go program scheduling or GC pause may cost too long.

The library [automaxprocs](https://github.com/uber-go/automaxprocs) has solved the problem that Go programs cannot set `GOMAXPROCS` correctly in the case of Containers. Should we introduce this library to make Prometheus work as expected in the Kubernetes environment, or is it not appropriate to import this library because of other considerations?

**Comment by brian-brazil** [comment]
This sounds like a feature request for Go, rather than something that every single application in existence should have to add.

**Comment by DavadDi** [comment]
@brian-brazil  Good catch. I have noticed the issue [33803](https://github.com/golang/go/issues/33803) that will be fixed on go 1.14. Indeed, It's better to be fixed in Golang. Good news for Gopher.

**Comment by fcantournet** [comment]
I think you can safely re-open this issue and include uber/automaxprocs
The issue is unlikely to be resolved at the runtime level any time soon.
It was moved to backlog (from 1.14) and it's actually not trivial to implement in `runtime` pkg itself since you have... well pretty much nothing.

NB: the env variable works, what doesn't is the default value the runtime chooses in the absence of this variable.
